### PR TITLE
Throw for empty strings in primitive collection columns

### LINKED ISSUE #32896: Query never returns when DB value for column mapped to primitive collection is ''
## Problem
When using EF to query an entity that has a primitive collection property the query never returns if the database value of the column is ''
The database query appears to execute successfully, but nothing more happens. The method never returns.
This is the last thing logged, even with LogLevel set to Trace:
```
info: 23.01.2024 14:47:32.803 RelationalEventId.CommandExecuted[20101] (Microsoft.EntityFrameworkCore.Database.Command)
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT "s"."Id", "s"."List"
      FROM "SimpleEntities" AS "s"
      LIMIT 1
```

Encountered this when adding a new column to an existing table. The generated migration sets the default value to ''
Tested with different collection types and primitives,  Sqlite/SqlServer, and sync/async. Same result for everything I tested with. 

Update: Left the repro running over night, the FirstAsync method has still not returned.

### Repro code
```C#
await using var connection = new SqliteConnection("DataSource=:memory:");
await connection.OpenAsync();

await using var ctx = new SimpleContext(connection);
await ctx.Database.EnsureDeletedAsync();
await ctx.Database.EnsureCreatedAsync();

await ctx.Database.ExecuteSqlRawAsync("INSERT INTO SimpleEntities (List) VALUES ('');");

var entity = await ctx.SimpleEntities.FirstAsync();
Console.WriteLine(entity);

public class SimpleContext(SqliteConnection connection) : DbContext
{
    public DbSet<SimpleEntity> SimpleEntities => Set<SimpleEntity>();

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder
            .UseSqlite(connection)
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();
    }
}

public record SimpleEntity(int Id, IEnumerable<string>? List);
```

### Include provider and version information

EF Core version: 8.0.1
Database provider: tested with Microsoft.EntityFrameworkCore.SqlServer and Microsoft.EntityFrameworkCore.Sqlite
Target framework: .NET 8.0)
Operating system: Windows 11
IDE: VS Code

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #32896

### CODE REVIEW

**Review by roji — COMMENTED** [review]
I'm not sure, but shouldn't we be throwing for an empty string, since it's invalid JSON?

**Comment by ajcvickers** [comment]
You're right; I didn't realize the empty string was invalid JSON.

**Comment by ajcvickers** [comment]
Updated to throw.

File: src/EFCore/Properties/CoreStrings.resx [hunk 1]
```diff
@@ -1,17 +1,17 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <root>
-  <!-- 
-    Microsoft ResX Schema 
-    
+  <!--
```
Comment (roji) [inline src/EFCore/Properties/CoreStrings.resx:3]: nit: unrelated whitespace changes (I get this too all of the time)

**Review by roji — APPROVED** [review]

File: src/EFCore/Properties/CoreStrings.resx (hunk 1 above)
Comment (ajcvickers) [reply src/EFCore/Properties/CoreStrings.resx:3]: Yeah, it's because the T4 that we generate from does not create code that conforms to our formatting, so whenever the file is opened in an editor, it changes from what was generated by T4.

Comment (roji) [reply src/EFCore/Properties/CoreStrings.resx:3]: Yeah, I remove the changes (`git checkout -p`) every time as well. It sucks.
